torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -561,7 +561,7 @@ TORCH_LIBRARY(_c10d_functional_autograd, m) {
       "SymInt[] output_split_sizes, "
       "SymInt[] input_split_sizes, "
       "str group_name) -> Tensor",
-      torch::dispatch(c10::DispatchKey::Autograd, ::all_to_all_single_autograd),
+      torch::dispatch(c10::DispatchKey::CompositeExplicitAutograd, ::all_to_all_single_autograd),
```

Comment:
Dumb q: does this work? I remember @d4l3k had some issues with it but I don't quite remember what.

Reply:
hmm, with the e2e test, I have to change it to `CompositeExplicitAutograd`. But I see what you meant as I saw there are tests failing. Let me double check this.